xarray/backends/scipy_.py

```diff
@@ -37,30 +37,39 @@ def __init__(self, variable_name, datastore):
 
     @property
     def array(self):
-        return self.datastore.ds.variables[self.variable_name].data
+        with self.datastore.ensure_open(autoclose=True):
+            return self.datastore.ds.variables[self.variable_name].data
 
     @property
     def dtype(self):
-        # always use native endianness
-        return np.dtype(self.array.dtype.kind + str(self.array.dtype.itemsize))
+        with self.datastore.ensure_open(autoclose=True):
```

Comment:
`NumpyIndexingAdapter` was just for overloading `__getitem__`. So we could simply move that into the `__getitem__` call instead, e.g., `NumpyIndexingAdapter(self.get_array())[key]`